src/CommitmentsV2/SFA.DAS.CommitmentsV2/Extensions/QueryableApprenticeshipsExtensions.cs

```diff
@@ -181,5 +186,102 @@ private static IQueryable<Apprenticeship> WithAlertsEmployer(this IQueryable<App
             return identifier.EmployerAccountId.HasValue ? 
                 apprenticeships.Where(app => app.Cohort.EmployerAccountId == identifier.EmployerAccountId) : apprenticeships;
         }
+
+        public static IQueryable<Apprenticeship> FilterApprenticeshipByAlert( IQueryable<Apprenticeship> apprenticeships, Alerts alert, long? providerId)
```

Comment:
The providerId appears to only be passed so that the code knows whether this is above providers or employers - is so then it should be a boolean flag.

Also this looks like it defaults to null as it's a nullable type and then default has been added to all the calls - this should be given a default value of null - so you don't have to pass all those default keywords above

Reply:
changed